pkg/pillar/cas/containerd.go

```diff
@@ -47,6 +47,51 @@ type containerdCAS struct {
 	ctrdClient *containerd.Client
 }
 
+// SnapshotUsage returns current usage of snapshot in bytes
+// if parents defined also adds all parents usage
+func (c *containerdCAS) SnapshotUsage(snapshotID string, parents bool) (int64, error) {
+	ctrdCtx, done := c.ctrdClient.CtrNewUserServicesCtx()
+	defer done()
+	var size int64
+	snapshots := []string{snapshotID}
+	if parents {
+		snapshotInfoList, err := c.ctrdClient.CtrListSnapshotInfo(ctrdCtx)
+		if err != nil {
+			return 0, fmt.Errorf("SnapshotUsage: Exception while fetching snapshots: %s. %s", snapshotID, err)
+		}
+		getParentID := func(snapshotID string) string {
+			for _, snap := range snapshotInfoList {
+				if snap.Name == snapshotID {
+					return snap.Parent
+				}
+			}
+			return ""
+		}
+		currentID := getParentID(snapshotID)
+		// iterate twice to not have possible loops
+		for range snapshotInfoList {
```

Comment:
It seems like you repeat the same work N times, where N is the number of members in snapshotInfiList. Why do you need to do that?

Reply:
I see you removed/rewrote this code, but can't tell what replaced it. Can you explain by responding to this comment?

Reply:
Sorry for the late respond. I slightly modified the code again.
The logic here is try to extract the chain of snapshots from the list, where provided snapshotID is the first one and it points onto another snapshot by `Parent` field. Another snapshot again has `Parent` field to point onto the third one. The last snapshot in the chain will have empty `Parent` field. I try to avoid possible circular dependency loops in the list of snapshots we get using `CtrListSnapshotInfo`. Not sure if circular dependencies are possible, but I do not want to be covered by issues in dependencies.